### DISCUSSION TITLE
Can i use ChromeOptions with undected mode?

### DISCUSSION THREAD

**Discussion by Jisan00b** [seleniumbase/SeleniumBase#3321] [General]
```python
import time
import logging
from selenium.webdriver import ChromeOptions, Chrome
from seleniumbase import Driver


def get_driver(use_base=False):
    chrome_options = ChromeOptions()
    prefs = {"download.default_directory": "downloads",
             "plugins.plugins_list": [{"enabled": False, "name": "Chrome Pdf Viewer"}],
             "plugins.always_open_pdf_externally": True,
             "profile.default_content_setting_values.automatic_downloads": 1,
             }
    chrome_options.add_argument('window-size=1200x600')
    chrome_options.add_argument("--no-sandbox")
    chrome_options.add_argument("--disable-setuid-sandbox")
    chrome_options.add_argument("--disable-dev-shm-usage")
    chrome_options.add_argument("--ignore-ssl-errors=yes")
    chrome_options.add_argument("--ignore-certificate-errors")
    chrome_options.add_experimental_option("prefs", prefs)
    chrome_options.accept_insecure_certs = True
    if use_base:
        logging.info("Instantiating Chrome with SeleniumBase...")
        return Driver(browser="chrome", undetected=True, options=chrome_options)
    else:
        logging.info("Instantiating Chrome with standard Selenium WebDriver...")
        return Chrome(options=chrome_options)


driver = get_driver()
driver.get("https://www.google.co.in/")
time.sleep(10)
driver.quit()

```

I'm trying to get undetected  driver with my custom options, is there any method which can get this done?

I saw the [discussions](https://github.com/seleniumbase/SeleniumBase/discussions/2346) which says both not possible. is any new method there OR in `Driver` can i add `prefs, --ignore-ssl-errors=yes & accept_insecure_certs` ?

**Comment by mdmintz** [comment]
Duplicate of https://github.com/seleniumbase/SeleniumBase/issues/1677#issuecomment-1382852136.

The important args you mentioned are already being set. Eg:

https://github.com/seleniumbase/SeleniumBase/blob/a353c53bb2783a02c556346fa2f32a56bc1d9090/seleniumbase/core/browser_launcher.py#L2210

https://github.com/seleniumbase/SeleniumBase/blob/a353c53bb2783a02c556346fa2f32a56bc1d9090/seleniumbase/core/browser_launcher.py#L2235

https://github.com/seleniumbase/SeleniumBase/blob/a353c53bb2783a02c556346fa2f32a56bc1d9090/seleniumbase/undetected/cdp_driver/config.py#L140

https://github.com/seleniumbase/SeleniumBase/blob/a353c53bb2783a02c556346fa2f32a56bc1d9090/seleniumbase/core/browser_launcher.py#L2203

**Reply by Jisan00b** [reply]
with `undetected=True` i want to set these:
```
    chrome_options.add_experimental_option("prefs", prefs)
    chrome_options.accept_insecure_certs = True
```

**Reply by mdmintz** [reply]
It should be the same as `--allow-running-insecure-content`, which is already included: https://github.com/seleniumbase/SeleniumBase/blob/a353c53bb2783a02c556346fa2f32a56bc1d9090/seleniumbase/core/browser_launcher.py#L2353

If you still think `options.accept_insecure_certs` does something different, I would need to see an example from you (with code), showing that setting `accept_insecure_certs` changes visible behavior.

**Reply by Jisan00b** [reply]
> with `undetected=True` i want to set these:
> 
> ```
>     chrome_options.add_experimental_option("prefs", prefs)
> ```

Brother i want to apply these experiment flags

>     prefs = {"download.default_directory": "downloads",
>              "plugins.plugins_list": [{"enabled": False, "name": "Chrome Pdf Viewer"}],
>              "plugins.always_open_pdf_externally": True,
>              "profile.default_content_setting_values.automatic_downloads": 1,
>              }

Here i can see`automatic_downloads` present  in `browser_launcher.py` & `always_open_pdf_externally` i can use by the flag `external_pdf`.

> "download.default_directory": my_external_dir,

But how can i pass download directory path to download.default_directory?

> "plugins.plugins_list": [{"enabled": False, "name": "Chrome Pdf Viewer"}],

And how to disable the Chrome Pdf Viewer plugin, i searched `plugins_list` found nothing

**Reply by mdmintz** [reply]
The `external_pdf=True` option lets you download PDFs instead of displaying them.
The download directory cannot be changed. (It is the `./downloaded_files/` folder.)
After files are downloaded, you can move them using Python.

**Reply by Jisan00b** [reply]
Got it, Thanks.
Is there any plan for making the download directory as argument instead of hardcoded path?
It would be helpful for downloading simultaneously from multiple instances in different directory.

**Reply by mdmintz** [reply]
It's hard-coded for a reason: All contents of the `./downloaded_files` folder are cleared out at the beginning of each `pytest` run. For safety, we won't let people set that folder to something else. People can use Python to copy files from that folder as needed.

**Reply by Jisan09** [reply]
Understandable. i'll have to fork & use for my own purpose, anyways thank you.
